From a4a59be600cbe17f43afa20422f9d05f3a5d5b52 Mon Sep 17 00:00:00 2001
From: Adrian Johnson <ajohnson@redneon.com>
Date: Sat, 7 Nov 2009 14:36:49 +1030
Subject: [PATCH] Type 1 subset: Don't add a new line when in binary mode

At the end of the private dictionary a new line is appended so that
the trailing zeros will not exceed 80 columns. In hex mode (used for
PS embedding) extra white space is harmless as it is ignored when
parsing the ascii hex encoded data. But in binary mode (used for PDF
embedding) the new line must not be added as it corrupts the binary
data.

https://bugs.launchpad.net/ubuntu/+source/cups/+bug/419143
---
 src/cairo-type1-subset.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/cairo-type1-subset.c b/src/cairo-type1-subset.c
index 8619e3e..b9e86c3 100644
--- a/src/cairo-type1-subset.c
+++ b/src/cairo-type1-subset.c
@@ -1100,7 +1100,8 @@ cairo_type1_font_subset_write_private_dict (cairo_type1_font_subset_t *font,
     if (status)
 	return status;
 
-    _cairo_output_stream_write (font->output, "\n", 1);
+    if (font->hex_encode)
+	_cairo_output_stream_write (font->output, "\n", 1);
 
     return CAIRO_STATUS_SUCCESS;
 }
-- 
2.1.0

